File: Sgp4Prop_v9.4/Sgp4Prop/SampleCode/Octave/DriverExamples/Sgp4Prop_Simple/Sgp4Prop_Simple.m
function Sgp4Prop_Simple()
  clc
  clear all
  %runs from the Sgp4Prop/SampleCode/Octave directory
  
  %OctFilePath=[pwd,"../../octfiles/lib"]; %autoload likes full paths
  dir=pwd;
  ndx=strfind(dir,"Octave");
  OctFilePath=dir(1:ndx+6);

  if(ispc)
    OctFilePath=[OctFilePath,"octfiles/lib/Windows/"]
  else %ismac and isunix are also options
    OctFilePath=[OctFilePath,"octfiles/lib/Linux/"]
  end
  FuncPath="../../octave_functions/";

  addpath(FuncPath)
  
  LoadAndInit(OctFilePath);
  line1 = "1 90021U RELEAS14 00051.47568104 +.00000184 +00000+0 +00000-4 0 0814"
  line2 = "2 90021   0.0222 182.4923 0000720  45.6036 131.8822  1.00271328 1199"
  satKey=TleAddSatFrLines(line1,line2);
  ierr=Sgp4InitSat(satKey);
  if(ierr!=0)
     msg=GetLastErrMsg()
  endif
  for mse = 0:1440:14400
    [ierr,ds50UTC,pos,vel,llh]=Sgp4PropMse(satKey,mse);
    printf("%6.0f  %12.6f %12.6f %12.6f %12.6f %12.6f %12.6f\n",mse,pos,vel)
    if(ierr!=0)
       msg=GetLastErrMsg()
    endif
  endfor
  
  ierr=TleRemoveSat(satKey);
  if(ierr!=0)
     msg=GetLastErrMsg()
  endif
  FreeLibs();
endfunction

function LoadAndInit(OctFilePath)
  loadDllMain(OctFilePath);
  loadEnvConst(OctFilePath);
  loadTimeFunc(OctFilePath);
  loadAstroFunc(OctFilePath);
  loadTle(OctFilePath);
  loadSgp4Prop(OctFilePath);
endfunction

function FreeLibs()
  freeSgp4Prop();
  freeTle();
  freeAstroFunc();
  freeTimeFunc();
  freeEnvConst();
  freeDllMain();
endfunction

File: Sgp4Prop_v9.4/Sgp4Prop/SampleCode/Julia/DriverExamples/Sgp4Prop/SGP4prop_simple.jl
#To run, add !#(path to julia) on the first line or run 
#(path to julia) ./julia_test.jl 

include("./AstroInit.jl")
using Printf

line1 = "1 90021U RELEAS14 00051.47568104 +.00000184 +00000+0 +00000-4 0 0814"
line2 = "2 90021   0.0222 182.4923 0000720  45.6036 131.8822  1.00271328 1199"

satKey=TleAddSatFrLines(line1, line2)
println(satKey)

iErrCode=Sgp4InitSat(satKey)

for mse in 0.0:2700.0:43200.0
	iErr, ds50UTC, pos, vel, llh=Sgp4PropMse(satKey, mse)
	yr,day=UTCToYrDays(ds50UTC)
        iErr, posNew, velNew, sgp4MeanKep=Sgp4PosVelToKep(yr,day,pos,vel)
	@printf " %17.7f%17.7f%17.7f%17.7f%17.7f%17.7f%17.7f\n" mse pos[1] pos[2] pos[3] vel[1] vel[2] vel[3]
        @printf " %17.7f%17.7f%17.7f%17.7f%17.7f%17.7f\n" sgp4MeanKep[1] sgp4MeanKep[2] sgp4MeanKep[3] sgp4MeanKep[4] sgp4MeanKep[5] sgp4MeanKep[6]
end

#infoStr=TimeFuncGetInfo()

#println(rstrip(infoStr))

#infomsg=GetLastInfoMsg()
#println(rstrip(infomsg))

